setup/helpers/abs_helper.rb

```diff
@@ -0,0 +1,657 @@
+        'volume_size': @metrics_volume_size }
+
+    hosts = [mom, metrics]
+
+    return hosts
+  end
+
+  # Attempts to provision the specified hosts via ABS
+  #
+  # hosts will likely come from abs_get_a2a_hosts:
+  #  {"mom": @abs_aws_mom_size, "metrics": @abs_aws_metrics_size}
+  #
+  # otherwise specify hosts in the following format
+  #  {"role1": "size1", "role2": "size2", ... }
+  #
+  # @author Bill Claytor
+  #
+  # @param [Array] hosts_to_request The hosts to request via ABS
+  #
+  # @return [JSON, nil] The hosts that were provisioned via ABS, otherwise nil
+  #
+  # @example
+  #   abs_get_resource_hosts = abs_get_resource_hosts(hosts_to_request)
+  #
+  def get_abs_resource_hosts(hosts_to_request)
+    hosts = []
+    abs_resource_hosts = nil
+    invalid_response = false
+
+    puts
+    puts "Attempting to provision ABS hosts: #{hosts_to_request}"
+    puts
+
+    # ensure the user has a token before proceeding
+    if abs_initialize
+      uri = URI("#{@abs_base_url}/awsdirect")
+      hosts_to_request.each do |host_to_request|
+        puts "host_to_request: #{host_to_request}"
+
+        role = host_to_request[:role]
+        size = host_to_request[:size]
+        volume_size = host_to_request[:volume_size]
+
+        request_body = get_awsdirect_request_body(role, size, volume_size)
+        response = perform_awsdirect_request(uri, request_body)
+
+        # if any invalid responses are encountered stop and return any provisioned hosts
+        # TODO: extract managing the responses and test separately
+        if !valid_abs_response?(response)
+          invalid_response = true
+          puts "Unable to provision host for role: #{role}"
+          puts
+
+          if hosts.empty?
+            puts "No ABS hosts were provisioned"
+            puts
+          else
+            puts "Returning any provisioned hosts"
+            return_abs_resource_hosts(hosts.to_json)
+          end
+
+          # stop requesting hosts
+          break
+
+        else
+          # reformat to satisfy beaker
+          hosts << JSON.parse(reformat_abs_resource_host(response.body))
+        end
+      end
+
+      if !hosts.empty? && !invalid_response
+        abs_resource_hosts = hosts.to_json
+        ENV["ABS_RESOURCE_HOSTS"] = abs_resource_hosts
+        puts "ABS_RESOURCE_HOSTS=#{ENV['ABS_RESOURCE_HOSTS']}"
+
+        # write to "last_abs_resource_hosts.log" (used when returning hosts)
+        update_last_abs_resource_hosts(abs_resource_hosts)
+      end
+
+    else
+      puts "Unable to proceed without a valid ABS token"
+      puts
+    end
+
+    # only return the hosts if they're successfully verified
+    success = verify_abs_hosts(hosts) unless abs_resource_hosts.nil?
+    verified_hosts = success ? abs_resource_hosts : nil
+
+    if verified_hosts
+      puts
+      puts "ABS hosts have been successfully provisioned"
+      puts "Pausing to ensure successful configuration"
+
+      # TODO: remove?
+      sleep 30
```

Comment:
Is this needed? Is there a way to test that configuration is successful instead of just arbitrarily sleeping?